README.md

```diff
@@ -38,7 +41,88 @@ Alpha version of API is available [here](http://docs.mir.dlang.io/latest/index.h
 - DMD (reference D compiler) >= `2.068`
 - LDC (LLVM D Compiler) `0.17.0`+.
 
-## Packages
+Rapid edit-run cycle
+--------------------
+
+The easiest way to execute your code is with `rdmd`.
+
+```
+rdmd -Isource examples/means_of_columns.d
+```
+
+`rdmd` is a companion to the `dmd` compiler that simplifies the typical edit-compile-link-run or edit-make-run cycle to a rapid edit-run cycle. Like make and other tools, `rdmd` uses the relative dates of the files involved to minimize the amount of work necessary. Unlike make, `rdmd` tracks dependencies and freshness without requiring additional information from the user.
+You can find more information [here](https://dlang.org/rdmd.html).
+
+Compile with ldc
+----------------
+
+The easiest way is to specify mir's sources during compilation:
+
+```
+ldc -Isource examples/means_of_columns.d
+```
+
+If you copy `mir`'s sources to `/usr/include/dlang/ldc/mir` you don't even need the `-Isource` include.
+You might want to have a look at dynamic linking with dmd below - it works too with `ldc`.
+
+For an additional performance boost, you can pass `-release -inline` to `ldc`.
+
+Compile with dmd
+----------------
+
+### Step 1: Compile your file(s)
+
+```
+dmd -c -Isource examples/means_of_columns.d
+```
+
+If you do this more often, you probably want to install a Mir package or put the mir sources to `/usr/include/dlang/dmd`.
+
+Now you can either use static linking (will copy everything into the binary) or dynamic linking (will load binary on run) to create an executable.
+
+### Step 2a: Static linking
+
+Static linking will result in copying all library routines in your binary. While this might require more disk space, it is faster and more portable.
+The only downside is that it requires recompilation if `mir` is updated.
+
+#### 2.a.1: Create static mir library
+
+You need to create a static library of `mir` once:
+
+```
+dmd -lib -oflibmir.a $(find source -name '*.d')
+```
+
+#### 2.a.2: Link statically
+
+```
+dmd means_of_columns.o libmir.a
+```
+
+### Step 2b: Dynamic linking
+
+With dynamic linking the OS will bind the binary and it's required external shared libraries at runtime.
+
+#### 2.b.1: Create shared mir library
+
+You need to create a shared, dynamic library of `mir` once:
+
+```
+dmd -shared -oflibmir.so -defaultlib=libphobos2.so -fPIC $(find source -name '*.d')
+```
+
+We need to specify:
+
+- `-defaultlib=libphobos2.so` as Phobos is statically linked by default.
+- `fPIC` to create Position Independent Code (it creates a global offset table)
+
+#### 2.b.2: Link dynamically
+
+```
+dmd means_of_columns.o -Llibmir.so -L-rpath=.
+```
+
+You can inspect the dynamic linking with `ldd means_of_columns`.
 
 ### On any platform with dub
```

Comment:
> Why between ldc and dmd ? Dub should be on top .

because unfortunately it's not that fast, so I definitely prefer `rdmd` over `dub`.